### PR TITLE
[MM-35617] App modals are now opened from components instead of actions

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR makes it so opening the app form modal and call response navigations are done by the components that call `doAppCall`, rather than `doAppCall` handling the modal and navigations. This is so the components can coordinate the transition with their own transition logic, post option's `closeWithAnimation` prop.

Components migrated to use this functionality:

- channel info bindings:
    - `screens/channel_info/bindings/bindings.tsx`

- channel post option bindings:
    - `screens/post_options/bindings/bindings.tsx`

- slash command submissions:
    - `actions/views/command.ts`
    - `draft_input.js`

- app form modal submissions:
    - `apps_form_container.tsx`
    - `apps_form_component.tsx`

- embedded forms:
    - `button_binding.tsx`
    - `menu_binding.tsx`

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-35617

#### Device Information

Android Pixel 2

iPhone simulator 12.3

#### Release Note

```release-note
NONE
```

Test app PR for testing navigation for the app form submission https://github.com/larkox/mattermost-app-test/pull/5

### CODE REVIEW

File: app/screens/post_options/bindings/bindings.tsx [hunk 1]
```diff
@@ -101,50 +105,59 @@ class Option extends React.PureComponent<OptionProps> {
...
-                    id: 'apps.error.unknown',
-                    defaultMessage: 'Unknown error occurred.',
-                });
-                Alert.alert(title, errorMessage);
-                return;
-            }
+        const callPromise = doAppCall(call, AppCallTypes.SUBMIT, intl);
+        await this.close();
+
+        const res = await callPromise;
```
Comment (mickmister) [inline app/screens/post_options/bindings/bindings.tsx:111]: You'll probably want to review this section with whitespace removed

**Review by larkox — APPROVED** [review]

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Tested various apps on various server for any issue opening the modal
- Tested opening a modal on a server when the client is connected via an ngrok tunnel
- Regression tested apps navigation function
- Regression tested general navigation functionality including permalinks and search
- Briefly regression tested other mobile functionality
- Tested on Android and iOS

LGTM!

Will retest if any code changes are request on the remain dev review.

**Comment by enahum** [comment]
@mickmister can you please resolve the conflicts? thanks

**Comment by mickmister** [comment]
@enahum Resolved conflicts

**Review by enahum — APPROVED** [review]

**Pull request merged**